jython_cli.java

```diff
@@ -0,0 +1,226 @@
+///usr/bin/env jbang "$0" "$@" ; exit $?
+
+//DEPS dev.jbang:jash:0.0.3
+//DEPS org.tomlj:tomlj:1.1.1
+//DEPS org.python:jython-slim:2.7.4
+//JAVA 21
+
+// jython-cli can be run with Java 8 from the command-line as follows:
+//
+// $ jbang --java 8 jython-cli
+//
+// [jbang] Building jar for jython_cli.java...
+//        Jython 2.7.4 (tags/v2.7.4:3f256f4a7, Aug 18 2024, 16:49:39)
+//        [OpenJDK 64-Bit Server VM (Temurin)] on java1.8.0_452
+//        Type "help", "copyright", "credits" or "license" for more information.
+//        >>>
+//
+// Java 8 is supported at least until the end of 2026
+// Thereafter, switch to Java 21
+// See https://adoptium.net/support/#_release_roadmap
+
+import java.io.*;
+import java.nio.file.*;
+import java.util.*;
+import java.util.stream.Stream;
+import dev.jbang.jash.Jash;
+import org.tomlj.Toml;
+import org.tomlj.TomlParseResult;
+import org.tomlj.TomlTable;
+
+import org.python.util.jython;
+
+public class jython_cli {
+
+    private static final String DEFAULT_JYTHON_VERSION = "2.7.4";
+    private static final String DEFAULT_JAVA_VERSION = "21";
+
+    // FIX_NUMBER appended to the Jython version forms the version of jython-cli as
+    // [Jython version].[FIX_NUMBER], e.g. 2.7.4.0
+    // Increment FIX_NUMBER with each new release of jython-cli.
+    // If the version number of Jython changes (for example 2.7.4 becomes 2.7.5), then
+    // FIX_NUMBER is reset to 0 again, e.g. 2.7.5.0
+    private static final int FIX_NUMBER = 0;  
+
+    private static final String textJythonApp = String.join(
+            "\n",
+            "",
+            "import org.python.util.jython;",
+            "",
+            "public class __CLASSNAME__ {",
+            "",
+            "    public static void main(String[] args) {",
+            "        System.setProperty(\"python.console.encoding\", \"UTF-8\");",
+            "        jython.main(args);",
+            "    }",
+            "",
+            "}",
+            "");
+
+    public static void main(String[] args) throws IOException {
+        List<String> deps = new ArrayList<>();
+        String jythonVersion = DEFAULT_JYTHON_VERSION;
+        String javaVersion = DEFAULT_JAVA_VERSION;
```

Comment:
What about Jeff's idea to get the versions from Java and Jython?
`System.getProperty("java.version")` / perhaps better [`Runtime.feature()`](https://apidia.net/java/OpenJDK/24/?pck=java.lang&cls=.Runtime#Version.feature-)
[`org.python.Version.PY_VERSION`](https://apidia.net/java/Jython/2.7.4/?pck=org.python&cls=.Version#PY_VERSION)